src/main/java/bongo/helper/Storage.java

```diff
@@ -0,0 +1,223 @@
+package bongo.helper;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.*;
+import java.time.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import bongo.task.Deadline;
+import bongo.task.Event;
+import bongo.task.Task;
+import bongo.task.Todo;
+
+/**
+ * A class for a Storage.
+ */
+public class Storage {
+
+    /**
+     * Different types of actions we can take to modify the text file.
+     */
+    public enum FileAction {
+        MARK_TASK,
+        UNMARK_TASK,
+        DELETE_TASK
+    }
+
+    private final String pathname;
+
+    /**
+     * A constructor for a Storage, with a pathname.
+     *
+     * @param pathname Pathname of file.
+     */
+    public Storage(String pathname) {
+        this.pathname = pathname;
+    }
+
+    /**
+     * Loads tasks into the TaskList, if there is an existing text file found.
+     * If no file is found, throws a FileNotFound Exception.
+     *
+     * @return An ArrayList of preloaded tasks.
+     * @throws FileNotFoundException If file is not found.
+     * @throws BongoException        If datetime string has an invalid format.
+     */
+    public ArrayList<Task> load() throws FileNotFoundException, BongoException {
+        File file = new File(this.pathname);
+        Scanner fileScanner = new Scanner(file);
+        ArrayList<Task> loadedTasks = new ArrayList<>();
```

Comment:
Good job naming the variable name for ArrayList with multiple items in plural form!